packages/react-components/react-carousel/library/src/components/useEmblaCarousel.ts

```diff
@@ -114,6 +114,53 @@ export function useEmblaCarousel(
     return plugins;
   }, [motion, onDragEvent, watchDrag]);
 
+  const reinitializeCarousel = React.useCallback(() => {
+    const plugins = getPlugins();
+
+    emblaOptions.current = {
+      startIndex: emblaOptions.current.startIndex,
+      align,
+      direction,
+      loop,
+      slidesToScroll,
+      watchDrag,
+      containScroll,
+    };
+
+    emblaApi.current?.reInit(
+      {
+        ...DEFAULT_EMBLA_OPTIONS,
+        ...emblaOptions.current,
+      },
+      plugins,
+    );
+  }, [align, containScroll, direction, getPlugins, loop, slidesToScroll, watchDrag]);
+
+  React.useEffect(() => {
+    reinitializeCarousel();
+  }, [reinitializeCarousel]);
```

Comment:
I don't see `reinitializeCarousel` being used anywhere, why this change is needed? Can we revert it?